apps/client/src/app/shop/[shop_id]/_states/server/api.ts

```diff
@@ -0,0 +1,21 @@
+import { axiosInstance } from "~/app/shared/server";
+import type {
+  GetNearestExpiryCouponResponse,
+  GetProductsByShopResponse,
+  GetShopDetailInfoResponse,
+} from "../../_types";
+
+export const API_SHOP_INFO = {
+  GetShopDetailInfo: async (shopId: number) => {
+    const res = await axiosInstance().get<GetShopDetailInfoResponse>(`/shops/${shopId}`);
+    return res.data.result;
+  },
+  GetNearestExpiryCoupon: async (shopId: number) => {
+    const res = await axiosInstance().get<GetNearestExpiryCouponResponse>(`/shops/${shopId}/coupons/shortest`);
+    return res.data.result;
+  },
+  GetProductsByShop: async (shopId: number) => {
+    const res = await axiosInstance().get<GetProductsByShopResponse>(`/${shopId}/products`);
+    return res.data.result.productList;
```

Comment:
클라이언트에 필요한 값이 응답 객체의 data 뿐이라면 api 작성은 이렇게 맞추는 것이 어떨까요?

```suggestion
  GetShopDetailInfo: async (shopId: number) => {
    const { data } = await axiosInstance().get<GetShopDetailInfoResponse>(`/shops/${shopId}`);
    return data.result;
  },
  GetNearestExpiryCoupon: async (shopId: number) => {
    const { data } = await axiosInstance().get<GetNearestExpiryCouponResponse>(`/shops/${shopId}/coupons/shortest`);
    return data.result;
  },
  GetProductsByShop: async (shopId: number) => {
    const { data } = await axiosInstance().get<GetProductsByShopResponse>(`/${shopId}/products`);
    return data.result
```

Reply:
@showme0241 
해당 부분은 #56 머지 완료되면 같이 수정 후 푸쉬하겠습니다!

Reply:
[chore: api 타입 변경](https://github.com/shophubkr/shophub-client/pull/52/commits/080d873b47852cf7d15c75cb39da5fe7e6137271)